jmclient/jmclient/blockchaininterface.py

```diff
@@ -408,18 +408,15 @@ def get_block_time(self, blockhash):
             return self.rpc('getblock', [blockhash])['time']
 
     def get_tx_merkle_branch(self, txid, blockhash=None):
-        """
-        Will fail and return None if the containing block has been pruned
-        """
         if not blockhash:
             tx = self.rpc("gettransaction", [txid])
             if tx["confirmations"] < 1:
-                return None
+                raise ValueError("Transaction not in block")
```

Comment:
A few more thoughts on this function: I think you need to check for `JsonRpcError` in this call too? Or does `gettransaction` never return error? I'm guessing it does. Also are you going to get a dict access error by calling `tx["confirmations"]` if the return value doesn't have it?
Also are you sure that a pruned blockchain is the only reason you can get an error returned from `gettxoutproof`? (I have no idea, just raising it, in case).
We're not catching `JsonRpcConnectionError` here but .. it seems we don't bother to catch it in most *other* places, other than `pushtx`, which is a bit weird on reflection.

Reply:
Those errors should be caught further up the call stack. Inside `get_tx_merkle_branch` we can't really do anything except raise another error. That function should only deal with expected errors like the block being pruned, unexpected errors like the bitcoin node going down should ideally be dealt with all in one place further up the call stack, otherwise you have repeated and less-maintainable code. The error handling also couldn't be general enough because a GUI would handle the error differently from CLI.

The return value of `gettransaction` always has `confirmations`: https://github.com/bitcoin/bitcoin/blob/60091d20f9bb6bb2d57b634567ba63b4f0e458d0/src/wallet/rpcwallet.cpp#L140